src/todoistSyncAPI.ts

```diff
@@ -165,7 +165,7 @@ export class TodoistSyncAPI   {
         //console.log(allActivity)
         const allActivityEvents = allActivity.events
         //client中不包含obsidian 的activity
-        const filteredArray = allActivityEvents.filter(obj => !obj.extra_data.client?.includes("obsidian")); 
+        const filteredArray = allActivityEvents.filter(obj => !obj.extra_data.client?.includes("obsidian") && this.plugin.settings.pullFromProjectId.includes(obj.parent_project_id) );
```

Comment:
If the judgment conditions in the latter part are added, it will result in some tasks created in Obsidian that cannot be synchronized back to Obsidian after being modified in Todoist.
```suggestion
        const filteredArray = allActivityEvents.filter(obj => !obj.extra_data.client?.includes("obsidian")
```

Reply:
Do i understand this correctly? This condition is not needed, if we implement a result4 object to select which tasks to create?